rktlet/runtime/helpers.go

```diff
@@ -202,7 +202,7 @@ func generateAppAddCommand(req *runtimeApi.CreateContainerRequest, imageID strin
 
 	// Add environments
 	for _, env := range config.Envs {
-		cmd = append(cmd, fmt.Sprintf("--environment=%s=%s", env.Key, env.Value))
+		cmd = append(cmd, fmt.Sprintf("--environment=%s='%s'", env.Key, strings.Replace(env.Value, `'`, `\'`, -1)))
```

Comment:
who is doing the unquoting? From what I can see in the rktlet code, this array will be used to call [exec.Command](https://golang.org/pkg/os/exec/#Command) directly on the rkt binary (I mean without `/bin/sh -c ...` intermediary process). So it looks like `/bin/sh` is not removing the quotes. Golang's `exec.Command()` will not do it either, see [stackoverflow](https://stackoverflow.com/questions/26473674/double-quotes-escaping-in-golang-exec).

So I suspect the unquoting is done either in rkt, or in systemd when parsing the unit file with the environments. From what I remember, bash and systemd don't have the same escaping rules (is it true?), so we need to know where the unquoting is done to know if this PR is correct.

And a comment should explain that ;)

Reply:
So, it is unquoted by the parser of `EnvironmentFile=` in systemd. According to the documentation, we should use double quotes: 
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=

> The parser strips leading and trailing whitespace from the values of assignments, unless you use double quotes (").

However, the implementation in  systemd's `parse_env_file_internal()` function allows simple quotes:
https://github.com/systemd/systemd/blob/v234/src/basic/fileio.c#L329-L334

So it works :)
But worth an explanation why, I think...

Reply:
Could you test with environment variable values containing both simple quotes and double quotes?

```
sudo rkt run --interactive docker://busybox "--environment=AAA='FOO'\"BAR'" --exec env | grep AAA
```

Reply:
I think that's a bug in rkt here ;)
rkt should add the quotes as appropriate when writing the environment file, and rktlet should not add the quotes.

Reply:
I'd agree this should be probably fixed in rkt directly.

Reply:
:+1: